### ISSUE TITLE
`JavaScriptStaticUrl.omitFunctionCall()` removes relation from graph.

### ISSUE THREAD

**Issue by kevincox** [assetgraph/assetgraph#1287]
I was surprised to find that calling `JavaScriptStaticUrl.omitFunctionCall()` removed the asset from the graph. I expected that it would remove the extra code to mark the asset but maintain the relation so that later actions such as renames would still affect that relation. Would it be possible to make it work that way?

**Comment by papandreou** [comment]
The relation should stay in the graph, it doesn't get removed here: https://github.com/assetgraph/assetgraph/blob/v7/lib/relations/JavaScriptStaticUrl.js#L25-L28

However, if you do something that causes the JavaScript asset to be serialized and reparsed (eg. by compressing it), the relation will "disappear" from the reparsed asset, because at that point it'll just be a string literal that we cannot detect as a relation.

That's why `assetgraph-builder` does it fairly late in the process: https://github.com/assetgraph/assetgraph-builder/blob/bc7c1a9d26647464f6e0030f9af66b27cd65bd49/lib/transforms/buildProduction.js#L541-L547

**Comment by kevincox** [comment]
Ah, you are probably right. I think I do this before compression. I'll try moving compression earlier and see if that helps. I guess it is mostly unavoidable that compression re-parses all of the relations.

**Comment by kevincox** [comment]
You are correct. I wonder if this can somehow be made more obvious to future developers. I guess at least people searching for this issue can find this ticket and learn from my mistakes.

**Comment by papandreou** [comment]
Yeah, I agree that it's a bit of a footgun. Hopefully some sort of standard annotation will land in the language soon :)
